co-simulation/fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/SumoAmbassador.java

```diff
@@ -162,6 +165,43 @@ protected void processInteractionAdvanced(Interaction interaction, long time) th
         }
     }
 
+    private void receiveInteraction(MsgerRequestTrafficEvent interaction) throws InternalFederateException {
+        
+        VehicleGetParameter veh = new VehicleGetParameter();
+        String temp  = "";
+        Set<String> vehicleData = traci.getSimulationControl().getKnownVehicles();
+        log.debug("Vehicles list: {}", vehicleData.toString());
+        try {
+            log.debug("Vehicle info: {}", interaction.toString());
+            if(!vehicleData.contains(interaction.vehicleId())){return;}
```

Comment:
Is this check really worth doing? Can we get an interaction for a vehicle that doesn't exist in SUMO? Seems like we're iterating over potentially a lot of vehicles for maybe not much benefit?